addon/components/layout/app-base/component.js

```diff
@@ -1,19 +1,22 @@
 import Ember from 'ember';
 import layout from './template';
 
-export default Ember.Component.extend({
+const { Component, inject } = Ember;
+
+export default Component.extend({
   layout,
 
-  currentUser: Ember.inject.service(),
+  currentUser: inject.service(),
   user: null,
   applicationLogo: null,
   indexRoute: 'index',
 
   didInsertElement() {
     this._super();
     this.get('currentUser').fetch().then(
-      ({ user }) => {
-        this.set('user', user);
+      ({ user, companies }) => {
+        this.set('user', Ember.Object.create(user));
```

Comment:
Instead of create a raw object, why not create a User object with in it some method like a computed for `fullName` for example ?